docs/odcs.rst

```diff
@@ -0,0 +1,120 @@
+
+API Changes
+-----------
+
+The following parameters will be used by ``buildContainer`` task defined by
+``koji-containerbuild`` plugin:
+
+``odcs_signed <bool, default True>`` When set to False, OSBS will not provide an
+empty list of ``sigkeys`` to ODCS causing the compose to potentially include
+unsigned packages. If ODCS integration is not in place, this parameter has no
+effect.
+
+``container.yaml`` Changes
+--------------------------
+
+In git repo, alongside Dockerfile, the ``container.yaml`` file will be enhanced
+to contain a "compose" section. This section will include all the required
+information for requesting a compose to be created by ODCS::
+
+    compose:
+      # Optional, default False.
+      inherit: true
+      # Optional, default "tag".
+      type: tag
+      # Required for "tag" type. Must contain at least one item.
+      packages:
+      - package-1-name # This is a package name, not an NVR.
+      - package-2-name
+      # Optional. Default value is configured in OSBS environment
+      sigkeys:
+      - AB123456
+
+Alternatively, for modules support::
+
+    compose:
+      type: module
+      # Eventually, ODCS will support ":" separator, for now "-" is used:
+      #     https://pagure.io/odcs/issue/98
+      # Multiple sources can be defined by separating them with a space.
+      # Required for "module" type.
+      source: "module_name1:stream1 module_name2:stream1"
+
+Behavior Changes
+----------------
+
+During an OSBS build, if ``container.yaml`` exists **and** ``container`` key is defined:
+
+- OSBS orchestrator build will request a compose from ODCS based on specified
+  parameters in ``container`` section. Once compose completes, build resumes and
+  the URL for the generated compose is passed to worker builds via
+  ``yum_repourls`` build parameter. *NOTE: If odcs-signed parameter was set to
+  false, the compose request will ignore the sigkeys provided in
+  container.yaml.*
+
+- If ``container`` section sets ``inherit`` to ``true``, the ODCS composes used
+  in parent image will be re-used. This may require OSBS to request compose to
+  be regenerated if it’s in *remove* state. Once compose is available, build
+  resumes and the URL for the generated compose is passed to worker builds via
+  ``yum_repourls`` build parameter. *NOTE: this can be used in conjunction with
+  the method stated on previous bullet point. In which case, multiple ODCS
+  composes are used.*
+
+- In the Koji build metadata, ``build.extra.image.odcs.compose_ids`` will be a
+  list of each ODCS compose used. Additionally,
+  ``build.extra.image.odcs.signed`` will correspond to the value of
+  ``odcs_signed`` build parameter which defaults to ``True``
+
+Otherwise, ODCS integration is completely bypassed. The previous behavior of
+using Koji yum repo for build tag of build target is used by default. If
+``yum-repo`` parameter is used, ODCS integration is also bypassed and only the
```

Comment:
Which parameter is this?

Reply:
I guess our various tools call this differently. I'll change it to `yum_repourls` to avoid confusion.

Reply:
Done.